### PR TITLE
[Bug] [Spark] Fix submit streaming task to spark engine not work

### LINKED ISSUE #3918: [Bug] [Spark] The configured streaming mode does not take effect when the spark engine is used
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

The configured streaming mode does not take effect when the spark engine is used

### SeaTunnel Version

2.3.0

### SeaTunnel Config

```conf
-
```

### Running Command

```shell
-
```

### Error Exception

```log
-
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

[](https://github.com/apache/incubator-seatunnel/issues/3918)

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TaoZex** [comment]
image

After the code is modified, if spark is set to job.mode=STREAMING, micro batch is used for processing.

**Comment by TyrantLucifer** [comment]
Could you please test it with kafka -> file case and offer the result of this test case? Result should include the following things:

1. Kafka topic offset whether update successfully.
2. Checkpoint whether trigger the logic that remove file from temp path to target path.
3. Verify the offset whether is same as the accutal row number of data be writed in file.

If these things are OK. this pr can be merged.

cc @CalvinKirs @Hisoka-X @EricJoy2048

**Comment by TaoZex** [comment]
> Could you please test it with kafka -> file case and offer the result of this test case? Result should include the following things:
> 
> 1. Kafka topic offset whether update successfully.
> 2. Checkpoint whether trigger the logic that remove file from temp path to target path.
> 3. Verify the offset whether is same as the accutal row number of data be writed in file.
> 
> If these things are OK. this pr can be merged.
> 
> cc @CalvinKirs @Hisoka-X @EricJoy2048


I have tested kafka source to file sink before and I will continue to test it as you suggest.

**Review by Hisoka-X — DISMISSED** [review]
LGTM

File: seatunnel-core/seatunnel-spark-starter/seatunnel-spark-3-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SinkExecuteProcessor.java [hunk 1]
```diff
@@ -120,11 +124,25 @@ public List<Dataset<Row>> execute(List<Dataset<Row>> upstreamDataStreams)
...
+                        .mode(SaveMode.Append)
+                        .save();
+            } else {
+                try {
+                    SparkSinkInjector.inject(dataset.writeStream(), seaTunnelSink)
+                            .option("checkpointLocation", "/tmp")
+                            .start()
+                            .awaitTermination();
+                } catch (StreamingQueryException | TimeoutException e) {
+                    throw new SeaTunnelException(
```
Comment (TyrantLucifer) [inline seatunnel-core/seatunnel-spark-starter/seatunnel-spark-3-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SinkExecuteProcessor.java:139]: SeaTunnelRuntimeException

**Review by hailin0 — DISMISSED** [review]

**Comment by Hisoka-X** [comment]
Hi, any progress? @TaoZex

**Comment by TaoZex** [comment]
> Hi, any progress? @TaoZex

Sorry, the past few days have been busy, and I will continue to finish this pr.

**Comment by TyrantLucifer** [comment]
Hi, @TaoZex How's it going? If you are too busy to complete this pull request, please let us know.

**Comment by laglangyue** [comment]
LSTM. And should the checkpoint location be configurable?

**Comment by TaoZex** [comment]
> Hi, @TaoZex How's it going? If you are too busy to complete this pull request, please let us know.

Sorry, I haven't had time to finish it because too busy lately. Last week I tested with kafka source-> file sink. Sometimes kafka data will fail to be written to file. If anyone are familiar with spark, please test it and give some suggestions.

**Comment by laglangyue** [comment]
I studied the relevant implementation of Spark today and found that the checkpoint should be set during context initialization using getOrCreateContext(), so I think this setting is incorrect.

It is reasonable to obtain the checkpoint path through SparkConf, which can be set at both task level and instance level through Spark lib.
Directly setting /tmp will result in no read and write permissions, especially in Spark on Yarn mode, where there are basically no permissions.

I will validate my ideas next week

**Comment by laglangyue** [comment]
Besides sparkcontext is in sparkEnvironment,it is Singleton,so we should set checkpoint when constructing it..

**Comment by laglangyue** [comment]
I find spark execution has the StreamContext objects，

I had a conversation with taozex，we will use struct streaming，not spark streaming.

StreamContext will be deleted

image

**Comment by hailin0** [comment]
Is there any progress?

**Comment by TaoZex** [comment]
> Is there any progress?

The task will continue through this pr:https://github.com/apache/incubator-seatunnel/pull/4713

**Comment by jeanleen** [comment]
any progress?

**Pull request closed without merging**